### ISSUE TITLE
Generic @property method can't be overridden in subclass

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1212]
**Describe the bug**
A `@property`-decorated method belonging to a generic class cannot be overridden in subclasses.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That overriding properties of generic classes would be allowed.

**Screenshots or Code**
```python
# pyright: strict

from typing import Generic, TypeVar

T = TypeVar('T')

class Foo(Generic[T]):
    def foo(self) -> T:
        ...

    @property
    def bar(self) -> T:
        ...

class Bar(Foo[str]):
    def foo(self) -> str:
        ...

    # "bar" overrides symbol of same name in class "Foo"
    #   Type "str" cannot be assigned to type "T"
    @property
    def bar(self) -> str:
        ...

class Baz(Foo[T]):
    # "bar" overrides symbol of same name in class "Foo"
    #   Type "T" cannot be assigned to type "T"
    @property
    def bar(self) -> T:
        ...

```

**VS Code extension or command-line**
Either, 1.1.89

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.90, which I just published. It will also be included in the next release of pylance.
